### PR TITLE
Added Textboxes, Re-oriented Overview tab, (issue-29-weekly-metrics)

### PR DESCRIPTION
**Pull request by nussbaum596**

### CODE REVIEW

File: data_import.R [hunk 1]
```diff
@@ -16,8 +16,8 @@ con <- DBI::dbConnect(
   driver   = "/Library/ODBC/FileMaker ODBC.bundle/Contents/MacOS/FileMaker ODBC",
   server   = "spsqlapwv003.sph.uthouston.edu",
   database = "DETECT",
-  uid      = keyring::key_list("detect_fm_db_readonly")[1,2],
-  pwd      = keyring::key_get("detect_fm_db_readonly")
+  uid      = 'brannuss',
+  pwd      = 'Badnnsbu#2021'
```
Comment (mbcann01) [inline data_import.R]: Please don't put your username and password directly in the code. Please use keyring instead.

File: index.Rmd [hunk 2]
```diff
@@ -273,62 +334,32 @@ summarise(
 valueBox(pct_change, icon = "fa-phone")
 ```
```
Comment (mbcann01) [inline index.Rmd]: This isn't giving the correct value. When I tested it, the dashboard says we made 215 calls two weeks ago, 192 calls last week, and that was a 89.3% change. But (215 -192) / 215 = 0.1069767 (about 11%).

File: index.Rmd [hunk 3]
```diff
@@ -273,62 +334,32 @@ summarise(
...
 
-
-
-Plots {data-width=750}
------------------------------------------------------------------------
-
-### Recruiting calls made and follow-up interviews scheduled by date
+### Total followup interviews made 2 weeks ago
```
Comment (mbcann01) [inline index.Rmd]: Please change to "Total follow-up interviews scheduled 2 weeks ago"

File: index.Rmd [hunk 4]
```diff
@@ -273,62 +334,32 @@ summarise(
...
 
-plotly::ggplotly(calls_per_day_plot)
+valueBox(total_followups_2_weeks, icon = "fa-phone")
 ```
 
-> The points on the plot above are colored according to the number of follow-up interviews scheduled on a given day.     
-> Red = None, Blue = 1, Green = 2 or more.
-
-### Ratio of number of scheduled F/U interviews to number of calls by day
+### Total followup interviews made last week
```
Comment (mbcann01) [inline index.Rmd]: Please change to "Total follow-up interviews scheduled last week"

File: dev_index.rmd [hunk 5]
```diff
@@ -79,3 +85,79 @@ summarise(
...
+    filter(week_num == x) %>%
+    select(n) %>%
+    max()
+  print(paste0("Highest Number of Calls Made During Week ", x, ": ", max_calls))
+}
+
+
+max_calls_func(2)
+
+```
```
Comment (mbcann01) [inline dev_index.rmd]: Sorry, the way I worded my request wasn't clear. What I meant was, "what is the highest number of calls we have made in any single week so far?"

For example, if we made 10 calls in week 1, 23 calls in week 2, and 18 calls in week, then the answer to the question above (and the value the textbox should display) is "23".

File: dev_index.rmd [hunk 6]
```diff
@@ -79,3 +85,79 @@ summarise(
...
+```{r}
+max_followups_func = function(x) {
+  max_followups = followups %>%
+  fill(week_num) %>%
+    filter(week_num == x) %>%
+    select(n) %>%
+    max()
+  print(paste0("Highest Number of Follow-Up Interviews Made During Week ", x, ": ", max_followups))
+}
+```
```
Comment (mbcann01) [inline dev_index.rmd]: Same comment as above for "Highest Number of calls made in any week"

**Review by mbcann01 — CHANGES_REQUESTED** [review]
Hi @nussbaum596,

I love the new row layout and the new boxes you created! Thank you!
I left some other specific comments below for you to address. Also, please go ahead and use index.Rmd for all of the textboxes. We are working on a development branch so we don't need to worry about messing up the actual dashboard.

Thank you!

**Comment by nussbaum596** [comment]
Dr. Cannell,

I will address these comments and get back the changes to you as soon as I can. Thanks.

BN

From: Brad Cannell ***@***.***>
Reply-To: brad-cannell/detect_recruitment_dashboard ***@***.***>
Date: Tuesday, April 27, 2021 at 7:09 PM
To: brad-cannell/detect_recruitment_dashboard ***@***.***>
Cc: nussbaum596 ***@***.***>, Mention ***@***.***>
Subject: Re: [brad-cannell/detect_recruitment_dashboard] Added Textboxes, Re-oriented Overview tab, (issue-29-weekly-metrics) (#32)


@mbcann01 requested changes on this pull request.

Hi @nussbaum596<https://github.com/nussbaum596>,

I love the new row layout and the new boxes you created! Thank you!
I left some other specific comments below for you to address. Also, please go ahead and use index.Rmd for all of the textboxes. We are working on a development branch so we don't need to worry about messing up the actual dashboard.

Thank you!

________________________________

In data_import.R<https://github.com/brad-cannell/detect_recruitment_dashboard/pull/32#discussion_r621684962>:

> -  uid      = keyring::key_list("detect_fm_db_readonly")[1,2],

-  pwd      = keyring::key_get("detect_fm_db_readonly")

+  uid      = 'brannuss',

+  pwd      = 'Badnnsbu#2021'

Please don't put your username and password directly in the code. Please use keyring instead.

________________________________

In index.Rmd<https://github.com/brad-cannell/detect_recruitment_dashboard/pull/32#discussion_r621705835>:

> @@ -273,62 +334,32 @@ summarise(

 valueBox(pct_change, icon = "fa-phone")

 ```

This isn't giving the correct value. When I tested it, the dashboard says we made 215 calls two weeks ago, 192 calls last week, and that was a 89.3% change. But (215 -192) / 215 = 0.1069767 (about 11%).

________________________________

In index.Rmd<https://github.com/brad-cannell/detect_recruitment_dashboard/pull/32#discussion_r621707255>:

> @@ -273,62 +334,32 @@ summarise(

 valueBox(pct_change, icon = "fa-phone")

 ```



-

-

-Plots {data-width=750}

------------------------------------------------------------------------

-

-### Recruiting calls made and follow-up interviews scheduled by date

+### Total followup interviews made 2 weeks ago

Please change to "Total follow-up interviews scheduled 2 weeks ago"

________________________________

In index.Rmd<https://github.com/brad-cannell/detect_recruitment_dashboard/pull/32#discussion_r621707376>:

>  ```



-> The points on the plot above are colored according to the number of follow-up interviews scheduled on a given day.

-> Red = None, Blue = 1, Green = 2 or more.

-

-### Ratio of number of scheduled F/U interviews to number of calls by day

+### Total followup interviews made last week

Please change to "Total follow-up interviews scheduled last week"

________________________________

In dev_index.rmd<https://github.com/brad-cannell/detect_recruitment_dashboard/pull/32#discussion_r621710797>:

> +### Highest Number of calls made in any week

+```{r}

+max_calls_func = function(x) {

+  max_calls = calls_per_day %>%

+  fill(week_num) %>%

+    filter(week_num == x) %>%

+    select(n) %>%

+    max()

+  print(paste0("Highest Number of Calls Made During Week ", x, ": ", max_calls))

+}

+

+

+max_calls_func(2)

+

+```

Sorry, the way I worded my request wasn't clear. What I meant was, "what is the highest number of calls we have made in any single week so far?"

For example, if we made 10 calls in week 1, 23 calls in week 2, and 18 calls in week, then the answer to the question above (and the value the textbox should display) is "23".

________________________________

In dev_index.rmd<https://github.com/brad-cannell/detect_recruitment_dashboard/pull/32#discussion_r621711035>:

> +### Highest Number of Follow-up Interviews Scheduled in Any week

+```{r}

+max_followups_func = function(x) {

+  max_followups = followups %>%

+  fill(week_num) %>%

+    filter(week_num == x) %>%

+    select(n) %>%

+    max()

+  print(paste0("Highest Number of Follow-Up Interviews Made During Week ", x, ": ", max_followups))

+}

+```

Same comment as above for "Highest Number of calls made in any week"

**Comment by nussbaum596** [comment]
Dr. Cannell,

I fixed the username/pw issue and fixed the issues with the other textboxes. However, I am struggling to get the plots and boxes oriented properly once I knit everything, even when playing around with the “data-height” numbers. Do you have any recommendations? Thank you, Dr. Cannell.

BN

From: Brad Cannell ***@***.***>
Reply-To: brad-cannell/detect_recruitment_dashboard ***@***.***>
Date: Tuesday, April 27, 2021 at 7:09 PM
To: brad-cannell/detect_recruitment_dashboard ***@***.***>
Cc: nussbaum596 ***@***.***>, Mention ***@***.***>
Subject: Re: [brad-cannell/detect_recruitment_dashboard] Added Textboxes, Re-oriented Overview tab, (issue-29-weekly-metrics) (#32)


@mbcann01 requested changes on this pull request.

Hi @nussbaum596<https://github.com/nussbaum596>,

I love the new row layout and the new boxes you created! Thank you!
I left some other specific comments below for you to address. Also, please go ahead and use index.Rmd for all of the textboxes. We are working on a development branch so we don't need to worry about messing up the actual dashboard.

Thank you!

________________________________

In data_import.R<https://github.com/brad-cannell/detect_recruitment_dashboard/pull/32#discussion_r621684962>:

> -  uid      = keyring::key_list("detect_fm_db_readonly")[1,2],

-  pwd      = keyring::key_get("detect_fm_db_readonly")

+  uid      = 'brannuss',

+  pwd      = 'Badnnsbu#2021'

Please don't put your username and password directly in the code. Please use keyring instead.

________________________________

In index.Rmd<https://github.com/brad-cannell/detect_recruitment_dashboard/pull/32#discussion_r621705835>:

> @@ -273,62 +334,32 @@ summarise(

 valueBox(pct_change, icon = "fa-phone")

 ```

This isn't giving the correct value. When I tested it, the dashboard says we made 215 calls two weeks ago, 192 calls last week, and that was a 89.3% change. But (215 -192) / 215 = 0.1069767 (about 11%).

________________________________

In index.Rmd<https://github.com/brad-cannell/detect_recruitment_dashboard/pull/32#discussion_r621707255>:

> @@ -273,62 +334,32 @@ summarise(

 valueBox(pct_change, icon = "fa-phone")

 ```



-

-

-Plots {data-width=750}

------------------------------------------------------------------------

-

-### Recruiting calls made and follow-up interviews scheduled by date

+### Total followup interviews made 2 weeks ago

Please change to "Total follow-up interviews scheduled 2 weeks ago"

________________________________

In index.Rmd<https://github.com/brad-cannell/detect_recruitment_dashboard/pull/32#discussion_r621707376>:

>  ```



-> The points on the plot above are colored according to the number of follow-up interviews scheduled on a given day.

-> Red = None, Blue = 1, Green = 2 or more.

-

-### Ratio of number of scheduled F/U interviews to number of calls by day

+### Total followup interviews made last week

Please change to "Total follow-up interviews scheduled last week"

________________________________

In dev_index.rmd<https://github.com/brad-cannell/detect_recruitment_dashboard/pull/32#discussion_r621710797>:

> +### Highest Number of calls made in any week

+```{r}

+max_calls_func = function(x) {

+  max_calls = calls_per_day %>%

+  fill(week_num) %>%

+    filter(week_num == x) %>%

+    select(n) %>%

+    max()

+  print(paste0("Highest Number of Calls Made During Week ", x, ": ", max_calls))

+}

+

+

+max_calls_func(2)

+

+```

Sorry, the way I worded my request wasn't clear. What I meant was, "what is the highest number of calls we have made in any single week so far?"

For example, if we made 10 calls in week 1, 23 calls in week 2, and 18 calls in week, then the answer to the question above (and the value the textbox should display) is "23".

________________________________

In dev_index.rmd<https://github.com/brad-cannell/detect_recruitment_dashboard/pull/32#discussion_r621711035>:

> +### Highest Number of Follow-up Interviews Scheduled in Any week

+```{r}

+max_followups_func = function(x) {

+  max_followups = followups %>%

+  fill(week_num) %>%

+    filter(week_num == x) %>%

+    select(n) %>%

+    max()

+  print(paste0("Highest Number of Follow-Up Interviews Made During Week ", x, ": ", max_followups))

+}

+```

Same comment as above for "Highest Number of calls made in any week"

**Comment by mbcann01** [comment]
Hi @nussbaum596 ,

I'm not exactly sure how to fix the issue you are dealing with off the top of my head, but I'm happy to take a look as soon as I get a chance. It probably won't be until next Thursday though.

Also, it looks like you are responding to these comments from your email, which is adding the entire email thread to the comment box on GitHub. If that is the case, do you mind replying directly on GitHub? It just helps keep the repo cleaner looking.

Thank you!

**Comment by mbcann01** [comment]
Left off on index.Rmd (test.Rmd)
- Moved the graph to the bottom of the page and text boxes to the top.
- I'm trying to get the layout to look nice (can I do columns in a row?). 
- Go through each of the comments above and make sure they are addressed.

File: index.Rmd (hunk 2 above)
Comment (mbcann01) [reply index.Rmd]: Fixed this.

**Comment by mbcann01** [comment]
Left off on index.Rmd (test.Rmd)
- Go through each of the comments above and make sure they are addressed.
- The last one I did was percent change in calls made.

**Pull request merged**